scriptler/findRunningPipelineJobs.groovy

```diff
@@ -0,0 +1,32 @@
+/*** BEGIN META {
+  "name" : "Find running pipeline jobs.",
+  "comment" : "Find all pipeline jobs running for at least a given amount of time (delay).",
+  "parameters" : [],
+  "core": "1.612.1",
+  "authors" : [
+    { name : "Arnaud Héritier" }
+  ]
+} END META**/
+
+import groovy.time.*
+
+def abortJobs = false
+
+use(TimeCategory)  {
+  
+  def delay = 1.day
+  
+  println "List of running jobs : "
+  Jenkins.instance.getAllItems(org.jenkinsci.plugins.workflow.job.WorkflowJob).each{
+    job -> job.builds.findAll{it.isBuilding() && new Date(it.startTimeInMillis) < ( new Date() - delay ) }.each{
```

Comment:
It is difficult to parameterize `delay` when using the TimeCategory.  I eliminated TimeCategory and took this approach:

```
envvar = this.binding  
  
/* Use existing value - if none, use provided default */
def getPropertyQuietly(envars, prop, defValue ) {
  try {
    if (envars.hasVariable(prop)) {
      // println "For ${prop} found '"+ envars.getVariable(prop) +"'"
      return envars.getVariable(prop)
    } else {
      return defValue
    }
  } catch (exception) {
     println "\nUsing default for $prop='$defValue'"
     println exception
     println "\n"
     return defValue     
  }
}  

def abortJobs = getPropertyQuietly(envvar, "abortJobs", 'false').toBoolean()
def oldJobs = 0
def maxAgeHours = getPropertyQuietly(envvar, "maxAgeHours", "12").toInteger()

  long delay = maxAgeHours * 60 * 60 * 1000
  def now = new Date();
  def cutOffTime = new Date(now.getTime() - delay)
```

This allows the value to be passed as a parameter from a job or when running the script.  Coincidentally, passing maxAgeHours as 0 lists all running Pipeline jobs.